prober/derp.go

```diff
@@ -577,6 +580,196 @@ func runDerpProbeNodePair(ctx context.Context, from, to *tailcfg.DERPNode, fromc
 	return nil
 }
 
+// derpProbeBandwidthMu ensures that derpProbeBandwidthTUN doesn't run concurrently.
+// This is necessary to avoid conflicts trying to create the TUN device.
+var derpProbeBandwidthMu sync.Mutex
+
+// derpProbeBandwidthTUN takes two DERP clients (fromc and toc) connected to two
+// DERP servers (from and to) and sends a test payload of a given size from one
+// to another over TUN.
+func derpProbeBandwidthTUN(ctx context.Context, transferTime *expvar.Float, from, to *tailcfg.DERPNode, fromc, toc *derphttp.Client, size int64) error {
+	derpProbeBandwidthMu.Lock()
+	defer derpProbeBandwidthMu.Unlock()
+
+	// Temporarily set up a TUN device with which to simulate a real client TCP connection
+	// tunneling over DERP.
+	dev, _, err := tstun.New(log.Printf, "utun9")
```

Comment:
Is this device name specific to a testing environment? Take a look at `tailscale/wireguard-go/tun.CreateTUN()` in `tun_darwin.go`.

Can we name it something more descriptive on Linux?

Reply:
It just has to start with `utun[0-9]`. How about `utun9_bwprobe` ?

Reply:
Darwin: If you omit the index suffix you will get one automatically assigned. Attempting to pin one assumes it's free. Anything other than [0-9] or omit will return an error.

For Linux, where this will run in practice, a name that includes `derpprobe` is helpful to distinguish from other TUN things, and `utun` has no meaning there.